core/request.py

```diff
@@ -309,44 +309,47 @@ def processRequest(self) -> None:
                     current.request.get().response.headers["Content-Type"] = "application/json"
-                    if isinstance(e, errors.HTTPException):
-                        res = {
-                            "status": e.status,
-                            "reason": e.name,
-                            "descr": str(translate(e.name)),
-                            "hint": e.descr,
-                        }
-                    else:
-                        res = {
-                            "status": 500,
-                            "reason": "Internal Server Error",
-                            "descr": descr
-                        }
-
-                    if conf["viur.instance.is_dev_server"]:
-                        res["traceback"] = traceback.format_exc()
+                    res = json.dumps(error_info)
 
-                    res = json.dumps(res)
+                else:  # We render the error in html
+                    # Try to get the template from html/error/
+                    if isinstance(e, errors.HTTPException):
+                        if conf["viur.instance.project_base_path"].joinpath(f"html/error/{e.status}.html").is_file():
+                            template = conf["viur.mainApp"].render.getEnv().get_template(
+                                conf["viur.mainApp"].render.getTemplateFileName(f"error/{e.status}"))
+                            res = template.render(error_info)
+                        elif conf["viur.instance.project_base_path"].joinpath("html/error/generic.html").is_file():
+                            template = conf["viur.mainApp"].render.getEnv().get_template(
+                                conf["viur.mainApp"].render.getTemplateFileName("error/generic.html"))
+                            res = template.render(error_info)
 
-                else:
-                    with (conf["viur.instance.core_base_path"].joinpath(conf["viur.errorTemplate"]).open() as tpl_file):
-                        tpl = Template(tpl_file.read())
-                        if isinstance(e, errors.HTTPException):
-                            res = tpl.safe_substitute({
-                                "error_code": e.status,
-                                "error_name": translate(e.name),
-                                "error_descr": e.descr,
-                            })
                         else:
-                            res = tpl.safe_substitute(
-                                {"error_code": "500",
-                                 "error_name": "Internal Server Error",
-                                 "error_descr": descr,
-                                 })
+                            with (conf["viur.instance.core_base_path"].joinpath(
+                                    conf["viur.errorTemplate"]).open() as tpl_file):
+                                template_file = Template(tpl_file.read())
+                                res = template_file.safe_substitute(error_info)
```

Comment:
You need to rename the variables in `error.html` too. However, IMO we should use here Jinja as renderer too.

Then you can move the template from `template/error.html` to `template/error/generic.html` and make the `elif conf["viur.instance.project_base_path"].joinpath("html/error/generic.html").is_file():` to the `else` block. If the generic template would be exist in the project folder, `getTemplateFileName()` would choose the template inside the core.

So the final code should look like 
```py
                        if conf["viur.instance.project_base_path"].joinpath(f"html/error/{e.status}.html").is_file():
                            template = conf["viur.mainApp"].render.getEnv().get_template(
                                conf["viur.mainApp"].render.getTemplateFileName(f"error/{e.status}"))
                            res = template.render(error_info)
                        else:
                            template = conf["viur.mainApp"].render.getEnv().get_template(
                                conf["viur.mainApp"].render.getTemplateFileName("error/generic.html"))
                            res = template.render(error_info)
```